controllers/scenario/scenario_controller_test.go

```diff
@@ -189,32 +175,38 @@ var _ = Describe("ScenarioController", func() {
 	})
 
 	It("can Reconcile function prepare the adapter and return the result of the reconcile handling operation", func() {
+		// I have no idea what is this magic but if either of resource deletion is removed, those test won't pass, erroring on:
+		// Message: "integrationtestscenarios.appstudio.redhat.com \"scenario-sample\" already exists"
+		Expect(k8sClient.Delete(ctx, hasScenario)).Should(Succeed())
 		result, err := scenarioReconciler.Reconcile(ctx, req)
 		Expect(reflect.TypeOf(result)).To(Equal(reflect.TypeOf(reconcile.Result{})))
 		Expect(err).To(BeNil())
 	})
 
 	It("can setup a new controller manager with the given reconciler", func() {
+		Expect(k8sClient.Delete(ctx, hasScenario)).Should(Succeed())
 		err := setupControllerWithManager(manager, scenarioReconciler)
 		Expect(err).NotTo(HaveOccurred())
 	})
 
 	It("Run reconciler for scenario", func() {
+		Expect(k8sClient.Delete(ctx, hasScenario)).Should(Succeed())
 		hasApp, err := scenarioReconciler.getApplicationFromScenario(ctx, failScenario)
 		Expect(err).NotTo(BeNil())
 		Expect(hasApp).To(BeNil())
 		Expect(err).To(HaveOccurred())
 	})
 
 	It("can fail when Reconcile fails to prepare the adapter when app is not found", func() {
-		Expect(k8sClient.Delete(ctx, failScenario)).Should(Succeed())
```

Comment:
why `failScenario` has been removed?

Reply:
Sorry, added back, wierd that it had no effect on passing the tests.